tests/main/xauth-migration/task.yaml

```diff
@@ -0,0 +1,83 @@
+summary: Check file XAUTHORITY env variable points to is migrated into the snap environment
+description: |
+    The XAUTHORITY environment variable points to a file which is located
+    in HOME, /run or /tmp depending on the distribution and desktop being
+    used. If the file ends up in /tmp then it wont be visible inside the
+    snap environment as each snap gets its private /tmp. To ensure the
+    file XAUTHORITY points to is always available no matter where it is
+    stored on the host system we migrate the file and copy it into
+    /run/$USER/.Xauthority and set the XAUTHORITY environment variable
+    accordingly.
+
+    This test verified the correct behaviour of the implemented logic
+    inside `snap run`.
+
+execute: |
+    snap install hello-world
+
+    ensure_xauth_path() {
+        export XAUTHORITY=$1
```

Comment:
(nitpick, sorry, shell makes my grumpy ;) - generally I think `export XAUTHORITY="$1"` (i.e. quotes) is better. However given that its a test etc this is nitpick^99 territory.